### PR TITLE
Add XML-implementation for ds:X509IssuerSerial

### PR DESCRIPTION
**Pull request by tvdijen**

### CODE REVIEW

File: src/XML/ds/X509IssuerSerial.php [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    public static function fromXML(DOMElement $xml): object
+    {
+        Assert::same($xml->localName, 'X509IssuerSerial', InvalidDOMElementException::class);
+        Assert::same($xml->namespaceURI, X509IssuerSerial::NS, InvalidDOMElementException::class);
+
+        $issuer = XMLUtils::extractStrings($xml, AbstractDsElement::NS, 'X509IssuerName');
+        $serial = XMLUtils::extractStrings($xml, AbstractDsElement::NS, 'X509SerialNumber');
+
+        Assert::minCount($issuer, 1, MissingElementException::class);
+        Assert::minCount($issuer, 1, TooManyElementsException::class);
```
Comment (jaimeperez) [inline src/XML/ds/X509IssuerSerial.php]: s/minCount/maxCount/?

File: src/XML/ds/X509IssuerSerial.php [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+        Assert::minCount($serial, 1, MissingElementException::class);
+        Assert::minCount($serial, 1, TooManyElementsException::class);
```
Comment (jaimeperez) [inline src/XML/ds/X509IssuerSerial.php]: Same here.

File: src/XML/ds/X509IssuerSerial.php [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+     */
+    public function toXML(DOMElement $parent = null): DOMElement
+    {
+        $e = $this->instantiateParentElement($parent);
+
+        XMLUtils::addString($e, AbstractDsElement::NS, 'X509IssuerName', $this->X509IssuerName);
+        XMLUtils::addString($e, AbstractDsElement::NS, 'X509SerialNumber', $this->X509SerialNumber);
+
+        // Fix the xs:type on the SerialNumber
+        $e->childNodes[1]->setAttributeNS(Constants::NS_XSI, 'xsi:type', 'xs:integer');
```
Comment (jaimeperez) [inline src/XML/ds/X509IssuerSerial.php]: This looks fragile. I assume `XMLUtils::addString()` does not return the element itself, right?

**Review by jaimeperez — CHANGES_REQUESTED** [review]

File: src/XML/ds/X509IssuerSerial.php (hunk 3 above)
Comment (tvdijen) [reply src/XML/ds/X509IssuerSerial.php:140]: It actually does.. I'll use that!

Comment (tvdijen) [reply src/XML/ds/X509IssuerSerial.php]: Nope, I'm open for suggestions

Comment (tvdijen) [reply src/XML/ds/X509IssuerSerial.php]: Well, the most easy solution would be to just add an XML-class for this of course! Even if it's a simple string-class... We can thoroughly test it, which we cannot now!

**Review by jaimeperez — APPROVED** [review]

**Pull request merged**